pyccel/parser/semantic.py

```diff
@@ -2978,6 +2978,8 @@ def _visit_PyccelSymbol(self, expr):
                 else:
                     errors.report(f"Unrecognised module {mod_name} imported in global scope. Please import the module locally if it was previously Pyccelised.",
                             severity='error', symbol = self.current_ast_node)
+            elif env_var is typing.Final:
+                return PyccelFunctionDef('Final', TypingFinal)
```

Comment:
What is this code about?

Reply:
This code allows me to do this:
https://github.com/pyccel/pyccel/blob/570b141bb6e8105c0eecf62f68858470509752ba/tests/epyccel/test_epyccel_lists.py#L877

The `Final` is parsed as the base of an `IndexedElement` but it still needs to be imported from the environment as the link to the import module is not obvious.

N.B if I had done:
```
 def list_equality(arg1 : typing.Final[list[int]], arg2 : typing.Final[list[int]]): 
```
then this change wouldn't be necessary, Pyccel would already recognise `typing` from the environment and use it to understand `Final`.

Reply:
I wanted to use `Final` in the tests as the wrapping for non-final lists is non-trivial and is therefore not implemented for Fortran.

We can either merge this as is because it is only 2 lines or I can move these 2 lines to #2303 and mark this PR as blocked.
Do you have a preference?

Reply:
How far is #2303 from being ready?